### PR TITLE
Private/tkrupa/cache encryption docs

### PR DESCRIPTION
**Pull request by tkrupa-intel**
### Details:
- Replacement for https://github.com/openvinotoolkit/openvino/pull/27993 (can't modify it, not from fork).
- Docs for https://github.com/openvinotoolkit/openvino/pull/27912. Added separately because docs addition breaks CI (no GPU Plugin in Python API tests currently),

### Tickets:
- CVS-158140

### CODE REVIEW

**Review by ilya-lavrenov — APPROVED** [review]

File: docs/articles_en/openvino-workflow/running-inference/optimize-inference/optimizing-latency/model-caching-overview.rst
Comment (sgolebiewski-intel) [inline docs/articles_en/openvino-workflow/running-inference/optimize-inference/optimizing-latency/model-caching-overview.rst:142]:
```suggestion
If model caching is enabled in the CPU Plugin, the model topology can be encrypted while it is saved to the cache and decrypted when it is loaded from the cache. This property can currently be set only in ``compile_model``.
```

File: docs/articles_en/openvino-workflow/running-inference/optimize-inference/optimizing-latency/model-caching-overview.rst
Comment (sgolebiewski-intel) [inline docs/articles_en/openvino-workflow/running-inference/optimize-inference/optimizing-latency/model-caching-overview.rst:160]:
```suggestion
If model caching is enabled in the GPU Plugin, the model topology can be encrypted while it is saved to the cache and decrypted when it is loaded from the cache. Full encryption only works when the ``CacheMode`` property is set to ``OPTIMIZE_SIZE``.
```

**Review by sgolebiewski-intel — APPROVED** [review]

**Comment by tkrupa-intel** [comment]
Removed from queue - the CI of this failed when this was added on top of https://github.com/openvinotoolkit/openvino/pull/27912. Let's wait until that gets merged first.

**Review by p-durandin — APPROVED** [review]

**Pull request merged**
